src/util/getNextThoughtsWithContextChain.js

```diff
@@ -0,0 +1,113 @@
+import { store } from '../store.js'
+
+import {
+  ROOT_TOKEN,
+  RANKED_ROOT
+} from '../constants.js'
+// util
+import {
+  head,
+  contextOf,
+  unroot,
+  getThoughtsRanked,
+  getThoughtsSorted,
+  getSortPreference,
+  isFunction,
+  meta,
+  pathToContext,
+  getContextsSortedAndRanked,
+  hashThought,
+  splitChain,
+  nextSibling as thoughtNextSibling
+} from '../util.js'
+import { isContextViewActive } from './isContextViewActive.js'
+
+const contextToThought = (context, thoughtIndex) => context.map(contextValue => {
+  const thought = thoughtIndex[hashThought(contextValue)]
```

Comment:
I recommend using `getThought` in cases like this. You should not be accessing `thoughtIndex` directly, but instead using the selectors and accessor methods.